### PR TITLE
Get rid of cornice hack

### PR DESCRIPTION
**Pull request by leplatrem**
That one took me a while :)

### CODE REVIEW

**Comment by leplatrem** [comment]
@ametaireau @natim r?

File: readinglist/resource.py [hunk 1]
```diff
@@ -76,6 +68,7 @@ class BaseResource(object):
     mapping = RessourceSchema()
     id_field = '_id'
     modified_field = 'last_modified'
+    schema_for = ('POST', 'PUT')
```
Comment (ametaireau) [inline readinglist/resource.py]: I'm not a big fan of `schema_for`, but I cannot find any better names…

- `validation_methods`
- `schema_validation_methods` ?

File: readinglist/resource.py [hunk 2]
```diff
@@ -85,6 +78,16 @@ def __init__(self, request):
         self.known_fields = [c.name for c in self.mapping.children]
         self.timestamp = self.db.last_collection_timestamp(**self.db_kwargs)
 
+    @property
+    def schema(self):
+        colander_schema = self.mapping
+
+        if self.request.method not in self.schema_for:
+            # No-op since payload is not validated against schema
+            colander_schema = colander.MappingSchema(unknown='preserve')
```
Comment (ametaireau) [inline readinglist/resource.py]: yes, this.

**Comment by ametaireau** [comment]
Neat, this sounds pretty good to me!

File: readinglist/resource.py (hunk 1 above)
Comment (leplatrem) [inline readinglist/resource.py]: Il like ``validation_methods``

@Natim ?

Comment (Natim) [inline readinglist/resource.py]: `validate_schema_for` ?

Comment (leplatrem) [inline readinglist/resource.py]: yes!

**Pull request merged**